superset/dashboards/dao.py

```diff
@@ -38,6 +38,20 @@ class DashboardDAO(BaseDAO):
     model_cls = Dashboard
     base_filter = DashboardFilter
 
+    @staticmethod
+    def get_dashboard(id_or_slug: str) -> Dashboard:
+        query = db.session.query(Dashboard).filter(
+            Dashboard.id_or_slug_filter(id_or_slug)
+        )
+        # Apply dashboard base filters
+        query = DashboardFilter("id", SQLAInterface(Dashboard, db.session)).apply(
+            query, None
+        )
+        dashboard = query.one_or_none()
+        if not dashboard:
```

Comment:
this `if` doesn't trigger when I enter a bogus slug or id and I'm not sure why yet